reactor-core/src/main/java/reactor/core/publisher/FluxGroupBy.java

```diff
@@ -751,7 +761,16 @@ void tryReplenish() {
 				produced = 0;
 				GroupByMain<?, K, V> main = parent;
 				if (main != null) {
-					main.s.request(p);
+					if (this.isFirstRequest) {
```

Comment:
should `isFirstRequest` be volatile? it seems it is updated in two different places that could be run on different threads

Reply:
Lol. Yeah, forgot to push them.

Reply:
> should `isFirstRequest` be volatile? it seems it is updated in two different places that could be run on different threads

It should not be volatile since it is done in 2 different path. One within drain Regular, another if there is fusión